### DISCUSSION TITLE
Constrained output with tokens added into the model

### DISCUSSION THREAD

**Discussion by denadai2** [outlines-dev/outlines#1353] [Q&A] [answered]
Dear community,

let's say I add new tokens into the model vocabulary. E.g. `[OUTLINES!]`. What happens with conditions like this one

```python
class Response(BaseModel):
    answer: constr(pattern='^\[OUTLINES!\](code_\d+){4}$')
```

Does the constraint take care about the existance of the additional tokens or does it constrain on the original model's vocabulary?

Thanks

**Comment by denadai2** [comment, marked as answer]
It outputs the special token correctly!

**Reply by rlouf** [reply]
This is very good news 😅
